x-pack/plugins/secops/public/components/timeline/body/index.tsx

```diff
@@ -4,138 +4,249 @@
-  background: ${props => (props.index % 2 !== 0 ? props.theme.eui.euiColorLightShade : 'inherit')};
+  background: ${props => (props.index % 2 === 0 ? props.theme.eui.euiColorLightShade : 'inherit')};
+  height: 100%;
+  max-width: ${props => props.maxWidth};
+  min-width: ${props => props.minWidth};
+  overflow: hidden;
+  overflow-wrap: break-word;
   padding: 5px;
 `;
 
 export const defaultWidth = 1090;
 
-const ExpandableDetails = styled.div`
-  width: 100%;
-`;
+const emptyNotes: Note[] = [];
 
 /** Renders the timeline body */
-export const Body = pure<Props>(
-  ({ columnHeaders, columnRenderers, data, height, id, rowRenderers }) => (
-    <ScrollableArea height={height} data-test-subj="scrollableArea">
-      {data.map(ecs => (
-        <Row key={ecs._id!}>
-          <TimeGutter />
-          {getRowRenderer(ecs, rowRenderers).renderRow(
-            ecs,
-            <>
-              <FlexRow>
-                <Pin type="pin" size="l" />
-                <DataDrivenColumns data-test-subj="dataDrivenColumns">
-                  {columnHeaders.map((header, index) => (
-                    <ColumnRender
-                      key={`cell-${header.id}`}
-                      data-test-subj="cellContainer"
-                      maxwidth={`${header.minWidth}px`}
-                      minwidth={`${header.minWidth}px`}
-                      index={index}
-                    >
-                      <Cell size="xs">
-                        {getColumnRenderer(header.id, columnRenderers, ecs).renderColumn(
-                          header.id,
-                          ecs
-                        )}
-                      </Cell>
-                    </ColumnRender>
-                  ))}
-                </DataDrivenColumns>
-              </FlexRow>
-              <FlexRow>
-                <ExpandableDetails data-test-subj="expandableDetails">
-                  <LazyAccordion
-                    id={`timeline-${id}-row-${ecs._id}`}
-                    buttonContent={`${JSON.stringify(ecs.event) || {}}`} // TODO: this should be `event.message` or `event._source`
-                    paddingSize="none"
-                  >
-                    <StatefulEventDetails data={ecs} />
-                  </LazyAccordion>
-                </ExpandableDetails>
-              </FlexRow>
-            </>
-          )}
-        </Row>
-      ))}
-    </ScrollableArea>
-  )
-);
+export class Body extends React.PureComponent<Props, State> {
+  public readonly state: State = {
+    expanded: {},
+    showNotes: {},
+  };
+
+  public render() {
+    const {
+      addNoteToEvent,
+      columnHeaders,
+      columnRenderers,
+      data,
+      height,
+      id,
+      notes,
+      onColumnSorted,
+      onFilterChange,
+      onPinEvent,
+      onUnPinEvent,
+      pinnedEventIds,
+      rowRenderers,
+      sort,
+      updateNote,
+    } = this.props;
+
+    const onToggleShowNotes = (eventId: string): (() => void) => () => {
```

Comment:
Optional - since you are using a class why not just create private functions in your class. To make the render a little simple to read